### PR TITLE
Add entity class limit explanation to entities and license references

### LINKED ISSUE #2434: Update entity limits with entity class limits
### Description
Entity limits, defined in a sensu license, originally only capped the total entity count. Entity class limits, defined in the sensu license as of 5.20.0, add additional limits for the count of entities in each entity class.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [x] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Affected pages
https://docs.sensu.io/sensu-go/latest/reference/license/#entity-limit (there could be other affected pages that I was not able to find).

### Expected behavior
Primarily, entity limits are defined by the total entity count. Secondarily, entity class limits are defined by entity count per entity class.

### Current behavior
Entity limits are defined by the total entity count.

### Corrections needed
Add some verbiage about entity class limits:
> Your commercial license includes the entity limit tied to your Sensu licensing package. An entity limit of 0 allows unlimited entities. Both agent and proxy entities count toward the overall entity limit. Contact Sensu to upgrade your commercial license.

### Additional context
See [proposal](https://docs.google.com/document/d/1lLphB_q2QE-sjhxd3EpQVW8AQiD4c17o5bjRJqR1Ymc/edit#heading=h.letafivq6wet) for more in-depth examples and user stories.

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
Adds explanation for entity class limits in entity and license references.

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2434

### CODE REVIEW

**Comment by hillaryfraley** [comment]
@nikkixdev A few questions...

1. It sounds like you'd only have agent class limits at this point, is that correct? I understand we might add more classes in the future, but since this is a new distinction that is less straightforward than the entity limit, I think it makes sense to explain it this way right now. What do you think?

2. Will entity class limits and class-based counts also be specified in the `sensuctl license info` or license API responses?
  - If so, I think I should update http://localhost:1313/sensu-go/5.20/reference/license/#view-entity-count-and-entity-limit with the complete response. I will need examples because I get "Error not found" when I run `sensuctl license info` or a GET call to the license API.
  - If not, where would users find their entity class limits and class-based counts?

**Comment by nikkictl** [comment]
> It sounds like you'd only have agent class limits at this point, is that correct? I understand we might add more classes in the future, but since this is a new distinction that is less straightforward than the entity limit, I think it makes sense to explain it this way right now. What do you think?

Entity class limits apply to both agent and proxy entities.

> Will entity class limits and class-based counts also be specified in the sensuctl license info or license API responses?

Not right now. This work will be done in https://github.com/sensu/sensu-enterprise-go/issues/1021, but that work is not planned for a future release yet.

> If so, I think I should update http://localhost:1313/sensu-go/5.20/reference/license/#view-entity-count-and-entity-limit with the complete response. I will need examples because I get "Error not found" when I run sensuctl license info or a GET call to the license API.
If not, where would users find their entity class limits and class-based counts?

Until https://github.com/sensu/sensu-enterprise-go/issues/1021 is completed, you should not need to update that information. Currently, users can find their entity class limits in their sensu license (yaml/json only), and their entity class counts by querying their store or querying sensuctl with a tool such as jq.

File: content/sensu-go/5.20/reference/entities.md [hunk 1]
```diff
@@ -38,6 +38,17 @@ All [commercial features][9] are available for free in the packaged Sensu Go dis
 If your Sensu instance includes more than 100 entities, [contact us][10] to learn how to upgrade your installation and increase your limit.
 See [the announcement on our blog][11] for more information about our usage policy.
 
+Commercial licenses include an entity limit and an entity class limit:
```
Comment (nikkictl) [inline content/sensu-go/5.20/reference/entities.md]: Entity limit and entity class limit are not required. If undefined, the limit does not exist. Multiple entity class limits can exist.
```suggestion
Commercial licenses may include an entity limit and entity class limits:
```

File: content/sensu-go/5.20/reference/entities.md [hunk 2]
```diff
@@ -38,6 +38,17 @@ All [commercial features][9] are available for free in the packaged Sensu Go dis
...
+
+- Entity limit: the maximum number of entities of all classes your license includes. Both agent and proxy entities count toward the overall entity limit.
+- Entity class limit: the maximum number of agent entities your license includes.
```
Comment (nikkictl) [inline content/sensu-go/5.20/reference/entities.md]:
```suggestion
- Entity class limits: the maximum number of entities of each entity class your license includes.
```

File: content/sensu-go/5.20/reference/license.md [hunk 3]
```diff
@@ -54,11 +54,21 @@ Error: not found
 
 ## Entity limit
 
-Your commercial license includes the entity limit tied to your Sensu licensing package.
+Your commercial license includes the entity limit and entity class limit tied to your Sensu licensing package.
```
Comment (nikkictl) [inline content/sensu-go/5.20/reference/license.md]:
```suggestion
Your commercial license may include the entity limit and entity class limits tied to your Sensu licensing package.
```

File: content/sensu-go/5.20/reference/entities.md (hunk 2 above)
Comment (hillaryfraley) [inline content/sensu-go/5.20/reference/entities.md]:
```suggestion
- Entity class limit: the maximum number of a specific class of entities (e.g. agent or proxy) that your license includes.
```

File: content/sensu-go/5.20/reference/license.md [hunk 4]
```diff
@@ -54,11 +54,21 @@ Error: not found
...
 An entity limit of `0` allows unlimited entities.
-Both agent and proxy entities count toward the overall entity limit.
 [Contact Sensu][8] to upgrade your commercial license.
 
+Your Sensu license includes two types of entity limits:
+
+- Entity limit: the maximum number of entities of all classes your license includes. Both agent and proxy entities count toward the overall entity limit.
+- Entity class limit: the maximum number of agent entities your license includes.
```
Comment (nikkictl) [inline content/sensu-go/5.20/reference/license.md]:
```suggestion
- Entity class limits: the maximum number of entities of each entity class your license includes.
```

Comment (hillaryfraley) [inline content/sensu-go/5.20/reference/license.md]:
```suggestion
- Entity class limit: the maximum number of a specific class of entities (e.g. agent or proxy) that your license includes.
```

**Review by nikkictl — CHANGES_REQUESTED** [review]

File: content/sensu-go/5.20/reference/entities.md [hunk 5]
```diff
@@ -38,6 +38,17 @@ All [commercial features][9] are available for free in the packaged Sensu Go dis
...
+
+For example, if your license has an entity limit of 10,000 and an agent entity class limit of 3,000, you cannot run more than 10,000 entities (agent and proxy) total.
+At the same time, you cannot run more than 3,000 agents.
+If you use only 1,500 agent entities, you can have 8,500 proxy entities before you reach the overall entity limit of 10,000.
+
+Use sensuctl or the license API to [view your entity counts and limits][29].
```
Comment (hillaryfraley) [inline content/sensu-go/5.20/reference/entities.md]:
```suggestion
Use sensuctl or the license API to [view your overall entity count and limit][29].
Your entity class limits are listed in your YAML/JSON Sensu license.
To find your entity class counts, query your store or sensuctl with a tool like [jq][30].
```

File: content/sensu-go/5.20/reference/entities.md [hunk 6]
```diff
@@ -1127,3 +1138,4 @@ spec:
 [26]: #processes-attributes
 [27]: ../agent/#discover-processes
 [28]: http://man7.org/linux/man-pages/man1/top.1.html
+[29]: ../license/#view-entity-count-and-entity-limit
```
Comment (hillaryfraley) [inline content/sensu-go/5.20/reference/entities.md:1141]:
```suggestion
[29]: ../license/#view-entity-count-and-entity-limit
[30]: https://stedolan.github.io/jq/
```

Comment (nikkictl) [reply content/sensu-go/5.20/reference/entities.md:1141]: Oh I'm not sure I would link `jq` here in our docs.

File: content/sensu-go/5.20/reference/entities.md [hunk 7]
```diff
@@ -1127,3 +1140,5 @@ spec:
...
+[30]: https://stedolan.github.io/jq/
```
Comment (hillaryfraley) [inline content/sensu-go/5.20/reference/entities.md]:
```suggestion
```

File: content/sensu-go/5.20/reference/entities.md [hunk 8]
```diff
@@ -38,6 +38,19 @@ All [commercial features][9] are available for free in the packaged Sensu Go dis
 If your Sensu instance includes more than 100 entities, [contact us][10] to learn how to upgrade your installation and increase your limit.
 See [the announcement on our blog][11] for more information about our usage policy.
 
+Commercial licenses may include an entity limit and entity class limits:
+
+- Entity limit: the maximum number of entities of all classes your license includes. Both agent and proxy entities count toward the overall entity limit.
+- Entity class limit: the maximum number of a specific class of entities (e.g. agent or proxy) that your license includes.
```
Comment (nikkictl) [inline content/sensu-go/5.20/reference/entities.md]:
```suggestion
- Entity class limits: the maximum number of a specific class of entities (e.g. agent or proxy) that your license includes.
```

File: content/sensu-go/5.20/reference/entities.md [hunk 9]
```diff
@@ -38,6 +38,19 @@ All [commercial features][9] are available for free in the packaged Sensu Go dis
...
+
+For example, if your license has an entity limit of 10,000 and an agent entity class limit of 3,000, you cannot run more than 10,000 entities (agent and proxy) total.
+At the same time, you cannot run more than 3,000 agents.
+If you use only 1,500 agent entities, you can have 8,500 proxy entities before you reach the overall entity limit of 10,000.
+
+Use sensuctl or the license API to [view your overall entity count and limit][29].
+Your entity class limits are listed in your YAML/JSON Sensu license.
+To find your entity class counts, query your store or sensuctl with a tool like [jq][30].
```
Comment (hillaryfraley) [inline content/sensu-go/5.20/reference/entities.md]:
```suggestion
To find your entity class counts, query your store or sensuctl with a tool like jq.
```

Comment (nikkictl) [inline content/sensu-go/5.20/reference/entities.md]: IMO we shouldn't document workarounds for features that don't exist yet. I think the issue history is sufficient enough for this. Might be best to remove these lines.

File: content/sensu-go/5.20/reference/entities.md (hunk 7 above)
Comment (nikkictl) [inline content/sensu-go/5.20/reference/entities.md]:
```suggestion
```

File: content/sensu-go/5.20/reference/license.md [hunk 10]
```diff
@@ -54,11 +54,21 @@ Error: not found
...
-Your commercial license includes the entity limit tied to your Sensu licensing package.
+Your commercial license may include the entity limit and entity class limits tied to your Sensu licensing package.
 An entity limit of `0` allows unlimited entities.
-Both agent and proxy entities count toward the overall entity limit.
 [Contact Sensu][8] to upgrade your commercial license.
 
+Your Sensu license includes two types of entity limits:
+
+- Entity limit: the maximum number of entities of all classes your license includes. Both agent and proxy entities count toward the overall entity limit.
+- Entity class limit: the maximum number of a specific class of entities (e.g. agent or proxy) that your license includes.
```
Comment (nikkictl) [inline content/sensu-go/5.20/reference/license.md]:
```suggestion
- Entity class limits: the maximum number of a specific class of entities (e.g. agent or proxy) that your license includes.
```

File: content/sensu-go/5.20/reference/entities.md [hunk 11]
```diff
@@ -38,6 +38,19 @@ All [commercial features][9] are available for free in the packaged Sensu Go dis
...
+- Entity limit: the maximum number of entities of all classes your license includes. Both agent and proxy entities count toward the overall entity limit.
+- Entity class limit: the maximum number of a specific class of entities (e.g. agent or proxy) that your license includes.
+
+For example, if your license has an entity limit of 10,000 and an agent entity class limit of 3,000, you cannot run more than 10,000 entities (agent and proxy) total.
+At the same time, you cannot run more than 3,000 agents.
+If you use only 1,500 agent entities, you can have 8,500 proxy entities before you reach the overall entity limit of 10,000.
+
+Use sensuctl or the license API to [view your overall entity count and limit][29].
+Your entity class limits are listed in your YAML/JSON Sensu license.
+To find your entity class counts, query your store or sensuctl with a tool like jq.
```
Comment (hillaryfraley) [inline content/sensu-go/5.20/reference/entities.md]:
```suggestion
```

File: content/sensu-go/5.20/reference/license.md [hunk 12]
```diff
@@ -54,11 +54,21 @@ Error: not found
 
 ## Entity limit
 
-Your commercial license includes the entity limit tied to your Sensu licensing package.
+Your commercial license may include the entity limit and entity class limits tied to your Sensu licensing package.
 An entity limit of `0` allows unlimited entities.
-Both agent and proxy entities count toward the overall entity limit.
 [Contact Sensu][8] to upgrade your commercial license.
 
+Your Sensu license includes two types of entity limits:
```
Comment (nikkictl) [inline content/sensu-go/5.20/reference/license.md]:
```suggestion
Your Sensu license may include two types of entity limits:
```

**Review by nikkictl — APPROVED** [review]
This is good to merge after that last change I mentioned!

**Pull request merged**